### PR TITLE
Fix issue $this

### PR DESCRIPTION
**Pull request by Dietermartens**
$this does not exist - changed to $contactId

### CODE REVIEW

**Comment by seamuslee001** [comment]
Jenkins ok to test

**Comment by Dietermartens** [comment]
Maybe i should give some more info about this change. Its reproducible if you want to edit a birthdate on a contact view. When you click on save, it keeps loading infinitely. If you watch the ajax calls then you see an error on line 155 with a $this thats not defined.

File: modules/civicrm_engage/civicrm_engage.module [hunk 1]
```diff
@@ -136,7 +136,7 @@ function civicrm_engage_civicrm_pageRun(&$page) {
     }
     $contactId = $page->getVar('_contactId');
     if (empty($contactId)) {
-      $contactId = CRM_Utils_Request::retrieve('cid', 'Positive', $this, TRUE, NULL, $_REQUEST);
+      $contactId = CRM_Utils_Request::retrieve('cid', 'Positive', $contactId, TRUE, NULL, $_REQUEST);
```
Comment (seamuslee001) [inline modules/civicrm_engage/civicrm_engage.module]: Looking at the code here https://github.com/civicrm/civicrm-core/blob/master/CRM/Utils/Request.php#L74 i think the more correct option might be to use $page i think

Comment (Dietermartens) [reply modules/civicrm_engage/civicrm_engage.module]: you're absolutely right!  I'll change that

**Pull request merged**